### ISSUE TITLE
Specify flavor in build time

### ISSUE THREAD

**Issue by Borlehandro** [yshrsmz/BuildKonfig#105]
Hi there! Are there any way to specify flavor in build time?
As I understand it, now plugin get flavor from properties only once when it's applied. But i want to change `buildkonfig.flavor` dynamically, for example based on my android-app flavor. I try to `ext.set("buildkonfig.flavor", <someFlavor>)` **before** `:generateBuildKonfig` task but it doesn't work for me.

**Comment by yshrsmz** [comment]
`buildkonfig.flavor` is just a property, so you can just pass it via gradlew command

```shell
 ./gradlew build -Pbuildkonfig.flavor=release
```

See detail here:
https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties

Also there's a note in README:
https://github.com/yshrsmz/BuildKonfig#product-flavor

**Comment by yshrsmz** [comment]
For the OP case, you can get an actual task name that a user passes to `gradlew` by `project.gradle.startParameter.taskNames`. So it's possible to dynamically change the value of `buildkonfig.flavor` property.

**Comment by yshrsmz** [comment]
For cocoapods, please open a separate issue as this one is already closed.
(And I'm not sure how Cocoapods & Kotlin Cocoapods Plugin build the project.)

**Comment by yshrsmz** [comment]
Or why not just update `buildkonfig.flavor` property with that `project.findProperty(KotlinCocoapodsPlugin.CONFIGURATION_PROPERTY)` value? I think that should  work.

**Comment by jadar** [comment]
Ah, that worked well. I put a `setProperty` based on that value I was using before, and now I can use `buildkonfig` as it was designed. Thanks!
